setup.py

```diff
@@ -496,6 +498,69 @@ def _read_requirements(filename: str) -> List[str]:
     return requirements
 
 
+def repackage_wheel(package_data: Dict[str, List[str]],
+                    wheel_location: str) -> None:
+    """Extracts libraries and other files from an existing wheel."""
+    assert _is_cuda(), "VLLM_USE_PRECOMPILED is only supported for CUDA builds"
+
+    import zipfile
+
+    if os.path.isfile(wheel_location):
+        wheel_path = wheel_location
+        print(f"Using existing wheel={wheel_path}")
+    else:
+        # Download the wheel from a given URL, assume
+        # the filename is the last part of the URL
+        wheel_filename = wheel_location.split("/")[-1]
+
+        import tempfile
+
+        # create a temporary directory to store the wheel
```

Comment:
we create a temp directory to hold the file, because right now all wheels have the same name, while people can try to install different versions.

Reply:
This will redownload the wheel on every install, as well as filling up `/tmp`, I think this will cause users issues in the long term.

Also, we should not have to have a workaround due to the choice of using the same wheel name for all releases, which breaks other scenarios (https://github.com/vllm-project/vllm/issues/9244)